apps/omg/lib/omg/state/transaction/validator.ex

```diff
@@ -50,7 +50,7 @@ defmodule OMG.State.Transaction.Validator do
          {:ok, outputs_spent} <- UtxoSet.get_by_inputs(utxos, inputs),
          :ok <- authorized?(outputs_spent, witnesses),
          {:ok, implicit_paid_fee_by_currency} <- Transaction.Protocol.can_apply?(raw_tx, outputs_spent),
-         true <- Fees.covered?(implicit_paid_fee_by_currency, fees) || {:error, :fees_not_covered} do
+         :ok <- Fees.check_if_covered(implicit_paid_fee_by_currency, fees) do
```

Comment:
`@type exec_error` needs `:overpaying_fees` and `:multiple_potential_currency_fees`?